cmd/kubernetesDeploy.go

```diff
@@ -0,0 +1,232 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+func kubernetesDeploy(myKubernetesDeployOptions kubernetesDeployOptions) error {
+	c := command.Command{}
+	// reroute stderr output to logging framework, stdout will be used for command interactions
+	c.Stderr(log.Entry().Writer())
+	runKubernetesDeploy(myKubernetesDeployOptions, &c, log.Entry().Writer())
+	return nil
+}
+
+func runKubernetesDeploy(myKubernetesDeployOptions kubernetesDeployOptions, command envExecRunner, stdout io.Writer) {
+	if myKubernetesDeployOptions.DeployTool == "helm" {
+		runHelmDeploy(myKubernetesDeployOptions, command, stdout)
+	} else {
+		runKubectlDeploy(myKubernetesDeployOptions, command)
+	}
+}
+
+func runHelmDeploy(myKubernetesDeployOptions kubernetesDeployOptions, command envExecRunner, stdout io.Writer) {
```

Comment:
Personally I dislike this long name `myKubernetesDeployOptions`, `options` would also do but up to you.

Reply:
you are right! 
The generator has recently been updated to suggest `config` as previously in Groovy.

I updated it accordingly ...